app/tutorials/example-applications/src/main/java/org/eclipse/mosaic/app/examples/emergencybrake/EmergencyBrakeApp.java

```diff
@@ -0,0 +1,217 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contact: mosaic@fokus.fraunhofer.de
+ */
+
+package org.eclipse.mosaic.app.examples.emergencybrake;
+
+import org.eclipse.mosaic.fed.application.ambassador.simulation.communication.CamBuilder;
+import org.eclipse.mosaic.fed.application.ambassador.simulation.communication.ReceivedAcknowledgement;
+import org.eclipse.mosaic.fed.application.ambassador.simulation.communication.ReceivedV2xMessage;
+import org.eclipse.mosaic.fed.application.app.ConfigurableApplication;
+import org.eclipse.mosaic.fed.application.app.api.CommunicationApplication;
+import org.eclipse.mosaic.fed.application.app.api.VehicleApplication;
+import org.eclipse.mosaic.fed.application.app.api.os.VehicleOperatingSystem;
+import org.eclipse.mosaic.interactions.communication.V2xMessageTransmission;
+import org.eclipse.mosaic.lib.enums.AdHocChannel;
+import org.eclipse.mosaic.lib.enums.SensorType;
+import org.eclipse.mosaic.lib.geo.GeoPoint;
+import org.eclipse.mosaic.lib.objects.v2x.MessageRouting;
+import org.eclipse.mosaic.lib.objects.v2x.etsi.Denm;
+import org.eclipse.mosaic.lib.objects.v2x.etsi.DenmContent;
+import org.eclipse.mosaic.lib.objects.vehicle.VehicleData;
+import org.eclipse.mosaic.lib.util.objects.ObjectInstantiation;
+import org.eclipse.mosaic.lib.util.scheduling.Event;
+import org.eclipse.mosaic.rti.TIME;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.File;
+
+/**
+ * This class implements an application for vehicles.
+ * In case the vehicle sensors detect an obstacle the vehicle will perform an emergency brake.
+ * If the emergency brake endures a specified minimum time duration a DENMessage is sent out.
+ */
+public class EmergencyBrakeApp extends ConfigurableApplication<CEmergencyBrakeApp, VehicleOperatingSystem> implements VehicleApplication,
+        CommunicationApplication {
+
+    CEmergencyBrakeApp configuration;
+
+    // Keep track of the vehicle movement for the emergency brake detection
+    private float lastSpeed = 0;
+    private float lastTime = 0;
+    private long startedBrakingAt = Long.MIN_VALUE;
+
+    // Keep status of the emergency brake performed on obstacle
+    private boolean emergencyBrake = false;
+    private long stoppedAt = Long.MIN_VALUE;
+
+    /**
+     * Initializes an instance of the {@link EmergencyBrakeApp}.
+     */
+    public EmergencyBrakeApp() { super(CEmergencyBrakeApp.class); }
+
+    @Override
+    public void onStartup() {
+        initConfig();
+        getOs().getAdHocModule().enable();
+    }
+
+    @Override
+    public void onShutdown() {
+
+    }
+
+    /**
+     * Sets all relevant variables from the configuration file.
+     * Note: If the file is corrupted or contains invalid values, default values are assigned.
+     */
+    public void initConfig() {
```

Comment:
You do not need to load the configuration by yourself anymore since you extended from `ConfigurableApplication`. You can delete this method and the field `configuration`, and use `getConfig()` instead whenever you need to access the configuration of the app.